### ISSUE TITLE
image configuration defaults

### ISSUE THREAD

**Issue by rneugeba** [moby/tool#16]
Looking at the YAML files in https://github.com/linuxkit/linuxkit there is an awful lot of repetition across them. We have discussed to simplify this with each package providing a template/default values which the user can override in the YAML file. In the best case the user then only needs to add two lines for a package either in the `onboot` or the `services` section containing the name and the reference to the image. This would greatly reduce the complexity of the YAML files while still allowing users to customise the settings to their needs.

It seems appropriate to store the default configuration in a label along with the image. The `moby` tool would then populate the internal representation of the package configuration from the label, before applying any overrides from the YAML file.

The proposal is to add a (optional) label `org.mobyproject.config` which contains a *subset* of the [image specification](https://github.com/linuxkit/linuxkit/blob/master/docs/yaml.md#image-specification) in YAML format, very much like they are currently present in the YAML files, but without the `name` and the `image` entries. The `moby` tool would take the label contents and override them with any entries specified in the YAML file before passing it to the validator and further processing.

An alternative would be to provide each entry as a separate label, but this seems to complicate processing.

## Example
Let's use the [dhcp](https://github.com/linuxkit/linuxkit/tree/master/pkg/dhcpcd) package as an example. The Dockerfile for the package would contain (may need some escaping of `"`s):
```
LABEL org.mobyproject.config="\
    binds:
     - /var:/var
     - /tmp/etc:/etc
    capabilities:
     - CAP_NET_ADMIN
     - CAP_NET_BIND_SERVICE
     - CAP_NET_RAW
    net: host
...
```

The a YAML file using the DHCP server in one-shot mode would look like:
```
...
onboot:
  - name: dhcpcd
    image: "linuxkit/dhcpcd:48e249ebef6a521eed886b3bce032db69fbb4afa"
    command: ["/sbin/dhcpcd", "--nobackground", "-f", "/dhcpcd.conf", "-1"]"
...
```
or if the user would like to start the `dhcpd` daemon as a service:
```
...
services:
  - name: dhcpcd
    image: "linuxkit/dhcpcd:48e249ebef6a521eed886b3bce032db69fbb4afa"
...
```

**Comment by justincormack** [comment]
I think there are issues with repository, commit and maintainer and would not add them. Lets focus on config.

**Comment by rneugeba** [comment]
I'm fine with focus on config first, but we must sooner rather than later also address the issue of being able to link a package to the source. I can edit the above text to focus on config and create a separate issue for the rest

**Comment by justincormack** [comment]
Its a very different issue, would rather do seperately

**Comment by rneugeba** [comment]
OK, updated the body

**Comment by SvenDowideit** [comment]
looks a lot like a default compose definition (as opposed to a moby specific thing), so I wouldn't call the label `org.mobyproject.config` - and instead something like `org.composevX.config` - where I would set `X` as low as possible for maximum utility.

**Comment by justincormack** [comment]
@SvenDowideit we have tried to avoid reference to Compose as the key names are different.

I think we would put the version in the actual yaml file not in the label name, but not 100% sure. We havent assigned any version yet as nothing is fixed and we are on 0.1 still...

Our labels dont exactly correspond to OCI at present either, as we have a few helpful simplifications. I kind of wish they did but its not that human friendly. Hmm.

**Comment by justincormack** [comment]
It turns out that its pretty much impossible to get new lines into Docker labels, via the CLI or Dockerfile, so going to try out using JSON.
